Source/DafnyCore/Compilers/Compiler-js.cs

```diff
@@ -1919,7 +1952,7 @@ protected class ClassWriter : IClassWriter {
 
     bool IsDirectlyComparable(Type t) {
       Contract.Requires(t != null);
-      return t.IsBoolType || t.IsCharType || AsNativeType(t) != null || t.IsRefType;
+      return t.IsBoolType || (t.IsCharType && !UnicodeChars) || AsNativeType(t) != null || t.IsRefType;
```

Comment:
This is because we translate to a CodePoint object, right?
But then I wonder: why don't we translate to a native int instead of an object?